RecoEgamma/EgammaHLTProducers/src/EgammaHLTBcHcalIsolationProducersRegional.cc

```diff
@@ -119,17 +138,23 @@ void EgammaHLTBcHcalIsolationProducersRegional::produce(edm::Event& iEvent, cons
       else
 	isol = isolAlgo.getTowerEtSum(&(*recoEcalCandRef));
 
-      if (doRhoCorrection_) {
-	if (fabs(recoEcalCandRef->superCluster()->eta()) < 1.442)
-	  isol = isol - rho*effectiveAreaBarrel_;
-	else
-	  isol = isol - rho*effectiveAreaEndcap_;
-      }
     } else { //calcuate H for H/E
       if (useSingleTower_)
 	isol = hcalHelper_->hcalESumDepth1BehindClusters(towersBehindCluster) + hcalHelper_->hcalESumDepth2BehindClusters(towersBehindCluster);
       else
 	isol = hcalHelper_->hcalESum(*(recoEcalCandRef->superCluster()));
+
+    }
+
+    if (doRhoCorrection_) {
+      int iEA = -1;
+      for (int bIt = absEtaLowEdges_.size() - 1; bIt > -1; bIt--) {
+        if ( fabs(recoEcalCandRef->superCluster()->eta()) > absEtaLowEdges_.at(bIt) ) {
```

Comment:
@afiqaize - please move the expensive fabs computation outside of the loop and cache it. Thanks!